transport-native-unix-common/src/main/c/netty_unix_socket.c

```diff
@@ -133,23 +133,24 @@ static jobject createDatagramSocketAddress(JNIEnv* env, const struct sockaddr_st
     return obj;
 }
 
-static jobject createDomainDatagramSocketAddress(JNIEnv* env, const struct sockaddr_storage* addr, int len, jobject local) {
-    jclass domainDatagramSocketAddressClass = NULL;
-    jobject obj  = NULL;
-    struct sockaddr_un* s = (struct sockaddr_un*) addr;
+static int domainSocketPathLength(const struct sockaddr_un* s, const socklen_t addrlen) {
 #ifdef __linux__
     // Linux supports abstract domain sockets so we need to handle it.
     // https://man7.org/linux/man-pages/man7/unix.7.html
-    int pathLength = 0;
     if (s->sun_path[0] == '\0') {
        // This is an abstract domain socket address
-       pathLength = strlen(&(s->sun_path[1])) + 1;
-    } else {
-       pathLength = strlen(s->sun_path);
+       return (addrlen - sizeof(sa_family_t));
     }
 #else
-    int pathLength = strlen(s->sun_path);
+    return strlen(s->sun_path);
 #endif
```

Comment:
```suggestion
#endif
    return strlen(s->sun_path);
```

Reply:
The diff above looks kind of weird. Basically, having `#else` is wrong. We should always unconditionally fall back to `strlen(s->sun_path)`